### PR TITLE
feat(store-ui): Add Skeleton Atom

### PR DESCRIPTION
**Pull request by lariciamota**
## What's the purpose of this pull request?
Add the Skeleton Atom to _store-ui_ with its story and test

## How to test it?
https://github.com/vtex-sites/storecomponents.store/pull/1197

### CODE REVIEW

**Comment by Gmantiqueira** [comment]
Awesome! Just one thing, I'm not sure about the styling. We're trying to decouple everything from @emotion and putting every CSS rule inside our themes. For example, the `theme-b2c-tailwind`. 

What about putting only the essential styles inside the `style={}` prop (Pure CSS), then everything else inside our theme? In this case, the problem are the keyframes solution with pure inline CSS.

Maybe something like this works:

```
let styleSheet = document.styleSheets[0];

let keyframes =
`@keyframes store-skeleton {
   0% {
    /* Animating absolute units, such as vw, has
    better performance than relative units, such as %,
    because it needs to be recalculated every frame */
    
    transform: translate3d(-150vw, 0, 0);
   }
   100% {
    transform: translate3d(0, 0, 0);
   }
 }`;

styleSheet.insertRule(keyframes, styleSheet.cssRules.length);
```

Then it'll push the keyframes rule with pure JS. But well, I'm not sure.

**Comment by lariciamota** [comment]
> Awesome! Just one thing, I'm not sure about the styling. We're trying to decouple everything from @emotion and putting every CSS rule inside our themes. For example, the `theme-b2c-tailwind`.
> 
> What about putting only the essential styles inside the `style={}` prop (Pure CSS), then everything else inside our theme? In this case, the problem are the keyframes solution with pure inline CSS.
> 
> Maybe something like this works:
> 
> ```
> let styleSheet = document.styleSheets[0];
> 
> let keyframes =
> `@keyframes store-skeleton {
>    0% {
>     /* Animating absolute units, such as vw, has
>     better performance than relative units, such as %,
>     because it needs to be recalculated every frame */
>     
>     transform: translate3d(-150vw, 0, 0);
>    }
>    100% {
>     transform: translate3d(0, 0, 0);
>    }
>  }`;
> 
> styleSheet.insertRule(keyframes, styleSheet.cssRules.length);
> ```
> 
> Then it'll push the keyframes rule with pure JS. But well, I'm not sure.

Good to know! Thank you, I'll look into it

**Comment by emersonlaurentino** [comment]
maybe you just leave the classes inside the component and the style is added by the theme. For example, if you are using Tailwind, you can use one of these animations: https://tailwindcss.com/docs/animation

File: packages/store-ui/src/atoms/Skeleton/Skeleton.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
+/** @jsxRuntime classic */
+/** @jsx jsx */
+import { forwardRef } from 'react'
+import type { HtmlHTMLAttributes } from 'react'
+import { jsx, css, keyframes } from '@emotion/core'
```
Comment (igorbrasileiro) [inline packages/store-ui/src/atoms/Skeleton/Skeleton.tsx:5]: We should use be using external frameworks?

File: packages/store-ui/src/atoms/Skeleton/Skeleton.tsx [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+export interface SkeletonProps extends HtmlHTMLAttributes<HTMLDivElement> {
```
Comment (igorbrasileiro) [inline packages/store-ui/src/atoms/Skeleton/Skeleton.tsx:7]: You're not using the HTMLAttributes<...> properties, can you remove it?

File: packages/store-ui/src/atoms/Skeleton/Skeleton.tsx (hunk 1 above)
Comment (igorbrasileiro) [reply packages/store-ui/src/atoms/Skeleton/Skeleton.tsx:5]: We can use styles directly.

**Review by igorbrasileiro — CHANGES_REQUESTED** [review]
Nice job @lariciamota 👏 ! Left some comments

**Comment by emersonlaurentino** [comment]
Hey, @lariciamota . I think this PR is a little confused, I say not in relation to what you did, but in relation to the Issue. 

We want our components to be style agnostic, just logic. However, it is impossible to make Skeleton without being biased by some lib (in your case @emotion/core). 

I believe we should review what exactly we want with this Component. One suggestion would be to have just one div that receives the `data-store-skeleton` and all the logic is in the `@vtex/theme-b2c-tailwind` theme. What do you think?

**Comment by Gmantiqueira** [comment]
> I believe we should review what exactly we want with this Component. One suggestion would be to have just one div that receives the `data-store-skeleton` and all the logic is in the `@vtex/theme-b2c-tailwind` theme. What do you think?

@emersonlaurentino I don't think so, this way our Skeleton component will be "useless" and the component functionality itself will be inside the theme. I prefer doing all the logic inside the component with native CSS, since the Skeleton doesn't need to have so many customizations. I believe that people will not change those styles very often.

The problem here is the CSS Specificity. If we put the logic inside the `style` attribute (inline), people will struggle to overwrite it.

File: packages/store-ui/src/atoms/Skeleton/Skeleton.tsx [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+}
+
+const Skeleton = forwardRef<HTMLDivElement, SkeletonProps>(function Skeleton(
+  { testId = 'store-skeleton', ...props },
+  ref
+) {
+  return (
+    <div ref={ref} data-store-skeleton-container data-testid={testId}>
+      <div data-store-skeleton {...props} />
+    </div>
```
Comment (emersonlaurentino) [inline packages/store-ui/src/atoms/Skeleton/Skeleton.tsx:17]: I like this approach, but what do you think about using only one div?

**Review by emersonlaurentino — APPROVED** [review]

File: packages/store-ui/src/atoms/Skeleton/Skeleton.tsx (hunk 3 above)
Comment (lariciamota) [reply packages/store-ui/src/atoms/Skeleton/Skeleton.tsx:17]: Yeah, it's better! I'll update it

File: packages/store-ui/src/atoms/Skeleton/Skeleton.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+export interface SkeletonProps {
+  /**
+   * ID to find this component in testing tools (e.g.: cypress, testing library, and jest).
+   */
+  testId?: string
+}
+
+const Skeleton = forwardRef<HTMLDivElement, SkeletonProps>(function Skeleton(
+  { testId = 'store-skeleton', ...props },
```
Comment (igorbrasileiro) [inline packages/store-ui/src/atoms/Skeleton/Skeleton.tsx:11]: You only have the testId props, maybe this isn't necessary.
But... If you want that your component receives all div props, you need to extend HTMLDivElement

File: packages/store-ui/src/atoms/Skeleton/stories/Skeleton.mdx [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+# Props
+
+<ArgsTable of={Skeleton} />
+
+# CSS Selectors
+
+```css
+[data-store-skeleton] {
+}
```
Comment (igorbrasileiro) [inline packages/store-ui/src/atoms/Skeleton/stories/Skeleton.mdx:18]:
```suggestion
[data-store-skeleton] {}
```

**Review by igorbrasileiro — APPROVED** [review]
Nice job!

File: packages/store-ui/src/atoms/Skeleton/stories/Skeleton.mdx [hunk 6]
```diff
@@ -0,0 +1,19 @@
+import { Story, Canvas, ArgsTable } from '@storybook/addon-docs'
+import Skeleton from '../Skeleton'
+
+# Preview
```
Comment (igorbrasileiro) [inline packages/store-ui/src/atoms/Skeleton/stories/Skeleton.mdx:4]:
```suggestion
# Skeleton
```

File: packages/store-ui/src/atoms/Skeleton/stories/Skeleton.mdx [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+
+<Canvas>
+  <Story id="atoms-skeleton--skeleton" />
+</Canvas>
+
+# Props
```
Comment (igorbrasileiro) [inline packages/store-ui/src/atoms/Skeleton/stories/Skeleton.mdx:10]:
```suggestion
## Props
```

File: packages/store-ui/src/atoms/Skeleton/stories/Skeleton.mdx [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+
+<ArgsTable of={Skeleton} />
+
+# CSS Selectors
```
Comment (igorbrasileiro) [inline packages/store-ui/src/atoms/Skeleton/stories/Skeleton.mdx:14]:
```suggestion
## CSS Selectors
```

**Review by Gmantiqueira — APPROVED** [review]
Nice!!

**Review by victorhmp — APPROVED** [review]

**Pull request merged**